docs/program/apis/disconnect.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Session Connectivity Resource Disonnection during a Session"
+linkTitle: "Session Connectivity"
+weight: 20
+type: "docs"
+description: "How to troubleshoot a resource losing connection during a robot session."
+tags: ["client", "sdk", "viam-server", "networking", "apis", "robot api", "session"]
+---
+
+A session:
+
+- allows a client to express that it is actively connected or authenticated to `viam-server` on your robot
+- supports stopping moving components when the connection is no longer active.
+
+### how does viam-server work if you lose connectivity during a session?
+
+If you lose connectivity, `viam-server` should stop this robot's session, therefore stopping the flow of power to all actuating resources.
+
+### how does the sdk work if you lose connectivity during a session?
+
+As a client of your robot, the Python or TypeScript SDKs will have their connection to the robot stopped when the robot loses connectivity, if the session is ended properly.
+Client implementation is necessary to make this work.
+
+### how the primary parts and sub parts work if you lose connectivity during a session?
+
+Resources on the robot will no longer be connected and signaling to each other if you lose connectivity and the session is ended.
+If opting to manage sessions yourself with the session management API, you must make sure the topographical hierachy of resources is respected when shutting down resources at session end.
+This means that sub-parts signal their shutdown to the primary part, which then closes the connection to accessing clients like SDKs.
```

Comment:
```suggestion
This means that you must ensure your sub-parts signal their shutdown to the primary part, which will then close the connection to accessing clients like SDKs.
```

Reply:
tense is weird on this one but i'm not sure what'd be best

Reply:
Sub-parts don't signal shutdown, as far as I know. Rather the session manager of the main part (that the client session is connected to) would send Stop() commands to the various resources. Again, I think there might be some conflation between "robot shutdown" and "end of session" here.

Reply:
ok agreed will try to rephrase, will comment once this is addressed

Reply:
Am confused here-- can I keep this in or is inaccurate? Tried to add more in earlier to un-conflate, but slightly unsure what to do here

Reply:
When a session ends, any in-progress actions are cancelled, generally causing all actuators to stop moving, typically by calling Stop(). This has nothing to do with topology or configuration. It's simply a way to make sure the robot doesn't keep driving/moving if the client connection (the thing sending commands) is lost.

During robot shutdown is where topographical ordering matters. A Close() is called on each part in the graph, starting from the "leaf" node resources (those with no dependencies) and working their way up to the robot itself. This shouldn't be related at all to just a network disconnection.

I don't think there's any need to talk about shutdown here, if you're only trying to document network disconnection behavior.